hermes/probe/create_file.go

```diff
@@ -0,0 +1,124 @@
+package probe
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	// "os"
+	// "sync"
+
+	// "cloud.google.com/go/storage"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	// "github.com/google/cloudprober/logger"
+	// "github.com/google/cloudprober/metrics"
+	// "github.com/google/cloudprober/probes/options"
+	// "github.com/google/cloudprober/targets/endpoint"
+	// "github.com/googleinterns/step224-2020/blob/master/hermes/hermes.go"
+)
+
+const (
+	begin = 1  // ID of the first HermesFile
+	end   = 50 // ID of the last HermesFile
+	maxFileSize = 1000 // maximum allowed file size in bytes 
+)
+
+type RandomHermesFile struct {
+	ID   int64 // ID is a positive integer in the range [1,50]
+	Size int   // File size in bytes
+}
+
+type randomHermesFileReader struct {
+	size int // size in bytes
+	i    int // currently reading this byte
+	rand *rand.Rand
+}
+
+func (r *randomHermesFileReader) readDone() bool {
```

Comment:
I think moving these conditions into methods is unnecessary complication and makes the reader jump back-and-forth. I would keep them in-line.

Reply:
It helped me a lot with 'studying' the code esp when debugging if that is alright I will leave that in when working on the code and will add a todo to move them back to inline?

Reply:
If not I am sorry and I will remove them now

Reply:
You are right that moving tricky condition expressions into well-named functions makes the code more readable.
However, this is rather simple code and in this case it makes it a bit harder to read as the reader has to jump back-and-forth. If you think the conditions don't speak for themselves, adding a clarifying comments would be better in this case.

Reply:
moved to inline and added clarifying comments